presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DetermineJoinDistributionType.java

```diff
@@ -170,8 +177,7 @@ public PlanNode visitDelete(DeleteNode node, RewriteContext<Void> context)
             }
 
             // Choose based on stats.
-            // We'll flip right and left later if the left side is smaller.
-            if (!mustPartitionJoin(node) && isSmall(node.getRight()) || !mustPartitionJoin(flipJoin(node)) && isSmall(node.getLeft())) {
+            if (isSmall(node.getRight()) && isMuchSmaller(node.getRight(), node.getLeft())) {
```

Comment:
I disagree. Here we're saying is the right node much smaller than the left node.